sklearn/neighbors/lof.py

```diff
@@ -167,8 +176,9 @@ def fit(self, X, y=None):
         self : object
             Returns self.
         """
-        if not (0. < self.contamination <= .5):
-            raise ValueError("contamination must be in (0, 0.5]")
+        if self.contamination != "auto":
+            if not(0. < self.contamination <= .5):
+                raise ValueError("contamination must be in (0, 0.5]")
```

Comment:
Not tested

Reply:
test added

Reply:
Please add the actual value of `self.contamination` in the error message:

```python
ValueError("contamination must be in (0, 0.5],"
           "got: %f" % self.contamination)
```